protocol/contracts/interfaces/templegold/ISpiceAuction.sol

```diff
@@ -0,0 +1,144 @@
+pragma solidity ^0.8.20;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Temple (interface/templegold/ISpiceAuction.sol)
+
+import { IAuctionBase } from "contracts/interfaces/templegold/IAuctionBase.sol";
+interface ISpiceAuction is IAuctionBase {
+    event AuctionConfigSet(uint256 epoch, SpiceAuctionConfig config);
+    event DaoExecutorSet(address daoExecutor);
+    event AuctionConfigRemoved(uint256 configId, uint256 epochId);
+    event LzReceiveExecutorGasSet(uint32 gas);
+    event RedeemedTempleGoldBurned(uint256 epochId, uint256 amount);
+    event OperatorSet(address indexed operator);
+
+    error InvalidConfigOperation();
+    error NotEnoughAuctionTokens();
+    error MissingAuctionTokenConfig();
+    error NoConfig();
+    error RemoveAuctionConfig();
+    error WithdrawFailed(uint256 amount);
+
+    struct SpiceAuctionConfig {
+        /// @notice Duration of auction
+        uint32 duration;
+        /// @notice Minimum time between successive auctions
+        uint32 waitPeriod;
+        /// @notice Cooldown after auction start is triggered, to allow deposits
+        uint32 startCooldown;
+        /// @notice Minimum Gold distributed to enable auction start
+        uint160 minimumDistributedAuctionToken;
+        /// @notice Address to start next auction when all criteria are met. Address zero means anyone can trigger start
+        address starter;
+        /// @notice Is Temple Gold auction token
+        bool isTempleGoldAuctionToken;
+        /// @notice Auction proceeds recipient
+        address recipient;
+    }
+
+    /// @notice Spice auction contracts are set up for 2 tokens. Either can be bid or sell token for a given auction
+    /// @notice Temple GOLD
+    function templeGold() external view returns (address);
+
+    /// @notice Spice Token
+    function spiceToken() external view returns (address);
+
+    /// @notice DAO contract to execute configurations update
+    function daoExecutor() external view returns (address);
+
+    /// @notice Operator
+    function operator() external view returns (address);
+
+    /// @notice Name of this Spice Bazaar auction
+    function name() external view returns (string memory);
+
+    /**
+     * @notice Set config for an epoch. This enables dynamic and multiple auctions especially for vested scenarios
+     * @dev Must be set before epoch auction starts
+     * @param _config Config to set
+     */
+    function setAuctionConfig(SpiceAuctionConfig calldata _config) external;
+
+    /// @notice Remove auction config set for last epoch
+    function removeAuctionConfig() external;
+
+    /**
+     * @notice Get auction token for current epoch
+     * @return Auction token
+     */
+    function getAuctionTokenForCurrentEpoch() external view returns (address);
+
+    /**
+     * @notice Get spice auction config for an auction
+     * @param auctionId Id of auction
+     */
+    function getAuctionConfig(uint256 auctionId) external view returns (SpiceAuctionConfig memory);
```

Comment:
A view function like `isActive()` that calls the `epochLib.isActive()` to immediately see if the current auction is active or not, would be very handy

Reply:
added